assets/src/data/eventespresso/base-entities.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import { upperFirst, camelCase, reduce, isMap, isNaN } from 'lodash';
+import pluralize from 'pluralize';
+import { mapReducer } from '@eventespresso/eejs';
+
+/**
+ * A helper for getting a method name.
+ * @param {string} modelName
+ * @param {string} suffix
+ * @param {string} prefix
+ * @param {boolean} usePlural
+ * @return {string} Returns a name for a method.
+ */
+export const getMethodName = (
+	modelName,
+	suffix = '',
+	prefix = 'get',
+	usePlural = false
+) => {
+	modelName = usePlural ? pluralize( modelName ) : modelName;
+	return prefix + upperFirst( camelCase( modelName + upperFirst( suffix ) ) );
+};
+
+/**
+ * Given a collection of existing entities and a collection of incoming
+ * entities, this returns a merged object with preference given to common
+ * entities from the existingEntities collection.
```

Comment:
> why favour existing over incoming?

Well for one that's the purpose of the method and why its named the way it is.  So for cases where this is needed (that existing entities have precedence) they'd call this method (which is how its needed in the places its being used).  So, no it can not be assumed that incoming entities are updated versions in some way **for the purpose and intended behaviour of this method**.